internal/test/e2e/version_compatibility_test.go

```diff
@@ -169,6 +171,107 @@ func TestVersionCompatibility(t *testing.T) {
 	})
 }
 
+func TestVersionCompatibility_PluginFieldUpdates(t *testing.T) {
+	cleanup := run.Koko(t)
+	defer cleanup()
+
+	dpCleanup := run.KongDP(kong.GetKongConfForShared())
+	defer dpCleanup()
+	require.NoError(t, util.WaitForKong(t))
+	require.NoError(t, util.WaitForKongAdminAPI(t))
+
+	kongAdmin, err := kongClient.NewClient(util.BasedKongAdminAPIAddr, nil)
+	require.NoError(t, err, "create go client for kong")
+	ctx := context.Background()
+	info, err := kongAdmin.Root(ctx)
+	require.NoError(t, err, "fetching Kong Gateway info")
+
+	dataPlaneVersion, err := versioning.NewVersion(kongClient.VersionFromInfo(info))
+	require.NoError(t, err)
+
+	admin := httpexpect.New(t, "http://localhost:3000")
+
+	// Remove plugins that may not be expected due to data plane version or aren't subject to compatibility updates
+	expectedPluginsMap := make(map[string]VersionCompatibilityPlugins, 0)
+	for _, plugin := range VersionCompatibilityOSSPluginConfigurationTests {
+		if plugin.FieldUpdateChecks == nil {
+			continue
+		}
+
+		if len(plugin.VersionRange) > 0 {
+			version := versioning.MustNewRange(plugin.VersionRange)
+			if !version(dataPlaneVersion) {
+				continue
+			}
+		}
+		expectedPluginsMap[plugin.Name] = plugin
+	}
+
+	// create plugins
+	for _, plugin := range expectedPluginsMap {
+		var config structpb.Struct
+		if len(plugin.Config) > 0 {
+			require.NoError(t, json.ProtoJSONUnmarshal([]byte(plugin.Config), &config))
+		}
+
+		p := &v1.Plugin{
+			Id:        uuid.NewString(),
+			Name:      plugin.Name,
+			Config:    &config,
+			Enabled:   wrapperspb.Bool(true),
+			Protocols: []string{"http", "https"},
+		}
+
+		pluginBytes, err := json.ProtoJSONMarshal(p)
+		require.NoError(t, err)
+		res := admin.POST("/v1/plugins").WithBytes(pluginBytes).Expect()
+		res.Status(http.StatusCreated)
+	}
+
+	// fetch list of plugins from the DP
+	dataPlanePlugins := []*kongClient.Plugin{}
+	util.WaitFunc(t, func() error {
+		dataPlanePlugins, err = kongAdmin.Plugins.ListAll(ctx)
+		if err != nil {
+			t.Log("fetch plugin failed", err)
+		}
+		if len(dataPlanePlugins) == 0 {
+			return errors.New("plugins len was 0")
+		}
+		return err
+	})
+
+	require.Equal(t, len(expectedPluginsMap), len(dataPlanePlugins), "plugins configured count does not match")
+	dpPluginsMap := make(map[string]*kongClient.Plugin, 0)
+	for _, dpPlugin := range dataPlanePlugins {
+		dpPluginsMap[*dpPlugin.Name] = dpPlugin
+	}
+
+	for name, expectedPlugin := range expectedPluginsMap {
+		t.Run(fmt.Sprintf("%s successfully updates fields", name), func(t *testing.T) {
+			require.Contains(t, dpPluginsMap, name, "plugin not present in the DP")
+			for rawVersion, updates := range expectedPlugin.FieldUpdateChecks {
+				parsedVersion := versioning.MustNewRange(rawVersion)
+				if parsedVersion(dataPlaneVersion) {
+					for _, update := range updates {
+						want := update.Value
+						have := dpPluginsMap[name].Config[update.Field]
```

Comment:
I just noticed this wont work when the Field value uses a path.to.field format. Eg: `update.Field = "response.body.time"`. I'll have to rewrite this using gjson or something...